junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedTestExtension.java

```diff
@@ -143,8 +148,22 @@ private ParameterizedTestNameFormatter createNameFormatter(Method templateMethod
 
 	private Object[] consumedArguments(Object[] arguments, ParameterizedTestMethodContext methodContext) {
 		int parameterCount = methodContext.getParameterCount();
-		return methodContext.hasAggregator() ? arguments
-				: (arguments.length > parameterCount ? Arrays.copyOf(arguments, parameterCount) : arguments);
+		if (methodContext.hasAggregator()) {
+			return arguments;
+		}
+		return arguments.length > parameterCount ? Arrays.copyOf(arguments, parameterCount) : arguments;
+	}
+
+	@SuppressWarnings("unchecked")
+	private List<Named<Object>> namedArguments(Object[] arguments) {
+		return Arrays.stream(arguments) //
+				.map(argument -> {
+					if (argument instanceof Named) {
+						return (Named<Object>) argument;
+					}
+					return Named.of(StringUtils.nullSafeToString(argument), argument);
```

Comment:
I think we can avoid this step here and continue to use `Object[]`. Otherwise, we'd have to handle exceptions here, too.